dev/update_requirements.py

```diff
@@ -0,0 +1,118 @@
+"""
+This script updates the `max_major_version` attribute of each package in a YAML dependencies
+specification (e.g. requirements/core-requirements.yaml) to the maximum available version on PyPI.
+"""
+
+import argparse
+import difflib
+import re
+
+import yaml
+import requests
+from packaging.version import Version, InvalidVersion
+
+
+def get_latest_major_version(package_name: str, minium_version: str = None) -> int:
+    url = f"https://pypi.org/pypi/{package_name}/json"
+    response = requests.get(url)
+    response.raise_for_status()
+    data = response.json()
+    min_ver = Version(minium_version if minium_version else "0.0.0")
+    versions = []
+    for version, distributions in data["releases"].items():
+        if len(distributions) == 0 or any(d.get("yanked", False) for d in distributions):
+            continue
+
+        try:
+            version = Version(version)
+        except InvalidVersion:
+            # Ignore invalid versions such as https://pypi.org/project/pytz/2004d
+            continue
+
+        if version.is_devrelease or version.is_prerelease or version < min_ver:
```

Comment:
Is comparison with `min_ver` necessary? The goal is to test the maximum possible major version, right? Is the ever a case where this version would be `< min_ver`?

Reply:
Good catch. We can remove it.

Reply:
Removed